scripts/extract_source.rb

```diff
@@ -253,10 +262,15 @@ def analyze_file(file)
             analysis.file_to_symbol_positions[cursor.location.file][cursor.spelling] = [start_offset, end_offset]
 
             cursor.visit_children do |child_cursor, parent|
+              # There seems to be a bug here on modern Clang versions where the
+              # cursor kind gets modified once we call "child_cursor.definition"
+              # - thus we make a copy ahead of calling that, for later use
+              child_cursor_kind = child_cursor.kind
```

Comment:
Oof. Is this worth reporting?

Reply:
Maybe? I think the issue here is likely in the Ruby binding, which is semi-maintained.

Its a bit of a time sink to reproduce this, so I'll punt on this for now.